pkg/util/util.go

```diff
@@ -8,3 +15,21 @@ func StringInSlice(str string, strs []string) bool {
 	}
 	return false
 }
+
+// ParseNodeSelectors parses the node selector string as a JSON array if possible, otherwise it treats it as a literal.
+func ParseNodeSelectors(selectors string) ([]string, error) {
+	selectors = strings.TrimSpace(selectors)
+
+	if len(selectors) == 0 {
+		return nil, errors.New("no node selector provided")
+	}
+
+	if strings.HasPrefix(selectors, "[") {
```

Comment:
Would [json.Valid](https://pkg.go.dev/encoding/json#Valid) be more reliable?
```suggestion
	if json.Valid([]byte(selectors)) {
```

Reply:
TIL about `json.Valid` 

I think we want to make sure that it's specifically an array.  We could parse the json and check, but that seems more complicated than trimming the leading whitespace and checking if it started with `[`.